packages/hash/frontend/.eslintrc.cjs

```diff
@@ -60,6 +56,12 @@ module.exports = {
               "Please use the custom src/components/hooks/useSnackbar hook instead.",
           },
         ],
+        patterns: [
+          {
+            group: ["@mui/material/*"],
+            message: "Please import from @mui/material instead",
+          },
+        ],
```

Comment:
Can this shadow other `patterns` from `@local/eslint-config`? If we have not packages that import from `@mui/material/*`, we can make this rule repo-wide.

Reply:
I did this, but it revealed that some rules in the base config were not previously having an effect in `design-system` – these ones:

```json
          {
            name: "@mui/material/Button",
            importNames: ["default"],
            message:
              "Please use the custom src/components/Button component instead.",
          },
```
– this can be fixed by removing `importNames: ["default"]`.

When I moved from `import Box from "@mui/material/Box"` (which was permitted as the above rule wasn't having an effect) to `import { Box } from "@mui/material"` I hit the rule that prevents that named import. 

The issue then becomes that there is no custom button in `design-system` to use – I think all of the 'please use x component in src/whatever' rules should apply to the `frontend` eslint config only. They also refer to file paths that do not exist (it should be `shared/ui` mostly). I'm guessing this is why it's 'legacy base to refactor', but given that moving the `@mui/material/*` rule has the knock-on effect of forcing that refactor, I suggest leaving it for this PR. I've reverted my branch.